### PR TITLE
pyTorch and onnxruntime backends and more

### PR DESCRIPTION
**Pull request by n-berezina-nn**
- support of pyTorch large models: kandinsky, falcon-7B, gpt-2, etc
- support of PyTorch compile backends
- support of onnxruntime tensor-RT backend
- support of validation for TF-lite c++ launcher
- support of random inputs instead of datasets
- separate binary for tf-lite GPU delegate

### CODE REVIEW

File: src/configs/README.md [hunk 1]
```diff
@@ -176,15 +176,70 @@
...
     потоков для выполнения вывода. По умолчанию будет выставлено число потоков, равное
     физическому количеству ядер в системе.
-  - `Delegate` - опциональный тег. Устанавливает путь до библиотеки-делегата, больше информации [здесь][https://www.tensorflow.org/lite/performance/delegates]. По умолчанию не установлен.
-  - `DelegateOptions` - опциональный тег. Устанавливает параметры для библиотеки-делегата в формате `option1: value1; option2: value2`.
+  - `Delegate` - опциональный тег. Устанавливает путь до библиотеки-делегата, больше информации
+    [здесь][https://www.tensorflow.org/lite/performance/delegates]. По умолчанию не установлен.
+  - `DelegateOptions` - опциональный тег. Устанавливает параметры для библиотеки-делегата в формате
+    `option1: value1; option2: value2`.
+
+- Набор тегов для тестирования вывода средствами PyTorch:
```
Comment (FenixFly) [inline src/configs/README.md:193]: Кажется здесь, либо в общих пропущен тег timeout_overhead

File: src/inference/README.md [hunk 2]
```diff
@@ -668,6 +669,108 @@ inference_pytorch.py
...
+```bash
+python inference_pytorch.py --model_name <model_name> \
+                            --input <path_to_data> \
+                            --input_name <input_name> \
+                            --input_shape <input_shape> \
+                            --mean <mean> --input_scale <scale> \
+                            --batch_size <batch_size>
+```
+
+**Запуск вывода в асинхронном режиме для модели, которая загружается из Gluon Model Zoo**
```
Comment (valentina-kustikova) [inline src/inference/README.md:729]: В описании примеров запуска pytorch откуда-то появились скрипты с mxnet. Надо удалить.

File: src/inference/configs/pytorch_configs/patches/1167.patch [hunk 3]
```diff
@@ -0,0 +1,58 @@
+From 109fee31c97ef334a682f06c976ff7c9bf25d87d Mon Sep 17 00:00:00 2001
```
Comment (valentina-kustikova) [inline src/inference/configs/pytorch_configs/patches/1167.patch:1]: Вроде бы таких файлов в репозитории не должно быть?

File: src/inference/inference_mxnet_sync_mode.py [hunk 4]
```diff
@@ -169,7 +169,7 @@ def inference_mxnet(net, num_iterations, get_slice, input_name, test_duration):
         mxnet.nd.waitall()
         t0 = time()
         slice_input = get_slice()
-        predictions = net(slice_input[input_name])
+        predictions = net(slice_input[input_name]).softmax()
```
Comment (valentina-kustikova) [inline src/inference/inference_mxnet_sync_mode.py:172]: Вроде бы обсуждали ранее, что softmax не должен входить в замер времени, а здесь откуда-то появился.

File: src/inference/inference_mxnet_sync_mode.py [hunk 5]
```diff
@@ -225,8 +225,18 @@ def main():
...
+            io.prepare_input(net, args.input)
+        else:
+            current_shape = model_wrapper.get_input_layer_shape(net, args.input_name)
+            transformed_shape = [
+                args.batch_size,
+                *io._transformer.get_shape_in_chw_order(current_shape, args.input_name[0]),
+            ]
+            custom_shapes = {args.input_name[0]: transformed_shape}
+            model_wrapper._input_shape = [transformed_shape]
+            io.fill_unset_inputs(net, log, custom_shapes)
```
Comment (valentina-kustikova) [inline src/inference/inference_mxnet_sync_mode.py:239]: В реализации асинхронного режима эта часть осталась без изменений.

File: src/inference/io_adapter.py [hunk 6]
```diff
@@ -396,23 +478,20 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
+
+        self.load_color_map('color_map.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:482]: Почему зашитое константное имя?

File: src/inference/io_adapter.py [hunk 7]
```diff
@@ -428,24 +507,22 @@ def process_output(self, result, log):
...
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:511]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 8]
```diff
@@ -461,23 +538,20 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
+
+        self.load_color_map('color_map_road_segmentation.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:542]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 9]
```diff
@@ -774,18 +848,16 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
+
+        self.load_labels_map('dictionary.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:852]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 10]
```diff
@@ -797,18 +869,11 @@ def process_output(self, result, log):
...
-                classes_color_map.append([int(x) for x in line.split()])
-        if not self._labels:
-            self._labels = os.path.join(os.path.dirname(__file__), 'labels/mscoco_names.txt')
-        labels_map = ['background']
-        with open(self._labels, 'r') as f:
-            for line in f:
-                labels_map.append(line.strip())
+
+        self.load_color_map('mscoco_color_map.txt')
+        self.load_labels_map('mscoco_names.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:874]: Аналогичные вопросы к этим двум строкам.

File: src/inference/io_adapter.py [hunk 11]
```diff
@@ -1511,18 +1576,17 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
-        if not self._labels:
-            self._labels = os.path.join(os.path.dirname(__file__), 'labels/kinetics.txt')
-        with open(self._labels, 'r') as f:
-            labels_map = [line.strip() for line in f]
+
+        self.load_labels_map('kinetics.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:1580]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 12]
```diff
@@ -1534,18 +1598,17 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
-        if not self._labels:
-            self._labels = os.path.join(os.path.dirname(__file__), 'labels/driver_action_labels.txt')
-        with open(self._labels, 'r') as f:
-            labels_map = [line.strip() for line in f]
+
+        self.load_labels_map('driver_action_labels.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:1602]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 13]
```diff
@@ -1557,18 +1620,9 @@ def process_output(self, result, log):
...
-                classes_color_map.append([int(x) for x in line.split()])
-        if not self._labels:
-            self._labels = os.path.join(os.path.dirname(__file__), 'labels/mscoco_names_90.txt')
-        labels_map = []
-        with open(self._labels, 'r') as f:
-            for line in f:
-                labels_map.append(line.strip())
+
+        self.load_color_map('mscoco_color_map_90.txt')
+        self.load_labels_map('mscoco_names_90.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:1625]: Аналогичный вопрос.

File: src/inference/io_adapter.py [hunk 14]
```diff
@@ -1738,10 +1792,9 @@ def process_output(self, result, log):
         if self._not_valid_result(result):
             log.warning('Model output is processed only for the number iteration = 1')
             return
-        if not self._labels:
-            self._labels = os.path.join(os.path.dirname(__file__), 'labels/pascal_voc.txt')
-        with open(self._labels, 'r') as f:
-            labels_map = [line.strip() for line in f]
+
+        self.load_labels_map('pascal_voc.txt')
```
Comment (valentina-kustikova) [inline src/inference/io_adapter.py:1796]: Аналогичный вопрос.

File: src/inference/configs/pytorch_configs/patches/1167.patch (hunk 3 above)
Comment (n-berezina-nn) [reply src/inference/configs/pytorch_configs/patches/1167.patch:1]: это workaround, чтобы заработала модель yolo-v7 в последней версии pytorch.
Патч можно будет убрать, когда эта проблема решится номинальным образом (мейнтейнеры репозитория сделают соответствующий фикс).

File: src/inference/io_adapter.py (hunk 10 above)
Comment (n-berezina-nn) [reply src/inference/io_adapter.py:874]: хардкод был изначально, сейчас убрали дублликации при загрузке этих файлов, поэтому они подсветились как измененные

File: src/inference/io_adapter.py (hunk 11 above)
Comment (n-berezina-nn) [reply src/inference/io_adapter.py:1580]: хардкод был изначально, сейчас убрали дублликации при загрузке этих файлов, поэтому они подсветились как измененные

File: src/inference/io_adapter.py (hunk 12 above)
Comment (n-berezina-nn) [reply src/inference/io_adapter.py:1602]: хардкод был изначально, сейчас убрали дублликации при загрузке этих файлов, поэтому они подсветились как измененные

File: src/inference/io_adapter.py (hunk 13 above)
Comment (n-berezina-nn) [reply src/inference/io_adapter.py:1625]: хардкод был изначально, сейчас убрали дублликации при загрузке этих файлов, поэтому они подсветились как измененные

File: src/inference/io_adapter.py (hunk 14 above)
Comment (n-berezina-nn) [reply src/inference/io_adapter.py:1796]: хардкод был изначально, сейчас убрали дублликации при загрузке этих файлов, поэтому они подсветились как измененные

File: src/inference/inference_mxnet_sync_mode.py (hunk 5 above)
Comment (n-berezina-nn) [reply src/inference/inference_mxnet_sync_mode.py:239]: fixed

**Review by maslovaz — DISMISSED** [review]

**Review by maslovaz — APPROVED** [review]

File: src/benchmark/frameworks/processes.py [hunk 15]
```diff
@@ -200,7 +199,7 @@ def __make_log_filename(self):
             test_settings.append(self._test.dep_parameters.runtime)
         if hasattr(self._test.dep_parameters, 'hint') and self._test.dep_parameters.hint:
             test_settings.append(self._test.dep_parameters.hint)
-        filename = '_'.join(test_settings) + '_' + datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+        filename = '_'.join(test_settings) + '_' + datetime.now().strftime('%Y-%m-%d_%H:%M:%S')
```
Comment (valentina-kustikova) [inline src/benchmark/frameworks/processes.py:202]: Здесь вроде бы Евгений писал, что двоеточия на Windows не работают как надо, из-за чего и заменили на дефисы. @FenixFly, я не ошибаюсь?

File: src/inference/README.md [hunk 16]
```diff
@@ -660,6 +661,86 @@ python inference_mxnet_sync_mode.py --model <file_name>.json \
...
+python inference_pytorch.py --model_name <model_name> \
+                            --input <path_to_data> \
+                            --input_name <input_name> \
+                            --input_shape <input_shape> \
+                            --mean <mean> --input_scale <scale> \
+                            --batch_size <batch_size>
+```
+
+
 ## Вывод глубоких моделей с использованием PyTorch (TorchVision)
```
Comment (valentina-kustikova) [inline src/inference/README.md:744]: Откуда-то два последовательных блока для pytorch. Здесь все в порядке?

**Review by valentina-kustikova — DISMISSED** [review]
Добавила пару комментариев, в остальном - вопросов нет.

File: src/benchmark/frameworks/processes.py (hunk 15 above)
Comment (n-berezina-nn) [reply src/benchmark/frameworks/processes.py:202]: да, это ошибка мержа, поправлю

**Review by valentina-kustikova — APPROVED** [review]

**Review by maslovaz — APPROVED** [review]

File: src/inference/README.md (hunk 16 above)
Comment (n-berezina-nn) [reply src/inference/README.md]: удалила один )

**Review by maslovaz — APPROVED** [review]
